providers/src/airflow/providers/amazon/aws/operators/dms.py

```diff
@@ -277,3 +288,551 @@ def execute(self, context: Context):
+        :ref:`howto/operator:DmsDeleteReplicationConfigOperator`
+
+    :param replication_config_arn: ARN of the replication config
+    :param wait_for_completion: If True, waits for the replication config to be deleted before returning.
+        If False, the operator will return immediately after the request is made.
+    :param deferrable: Run the operator in deferrable mode.
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+        If this is ``None`` or empty then the default boto3 behaviour is used. If
+        running Airflow in a distributed manner and aws_conn_id is None or
+        empty, then default boto3 configuration would be used (and must be
+    """
+
+    aws_hook_class = DmsHook
+    template_fields: Sequence[str] = aws_template_fields("replication_config_arn")
+
+    VALID_STATES = ["failed", "stopped", "created"]
+    DELETING_STATES = ["deleting"]
+    TERMINAL_PROVISION_STATES = ["deprovisioned", ""]
+
+    def __init__(
+        self,
+        *,
+        replication_config_arn: str,
+        wait_for_completion: bool = True,
+        deferrable: bool = conf.getboolean("operators", "default_deferrable", fallback=False),
+        waiter_delay: int = 5,
+        waiter_max_attempts: int = 60,
+        aws_conn_id: str | None = "aws_default",
+        **kwargs,
+    ):
+        super().__init__(
+            aws_conn_id=aws_conn_id,
+            **kwargs,
+        )
+
+        self.replication_config_arn = replication_config_arn
+        self.wait_for_completion = wait_for_completion
+        self.deferrable = deferrable
+        self.waiter_delay = waiter_delay
+        self.waiter_max_attempts = waiter_max_attempts
+
+    def execute(self, context: Context) -> None:
+        results = self.hook.describe_replications(
+            filters=[{"Name": "replication-config-arn", "Values": [self.replication_config_arn]}]
+        )
+
+        if len(results) > 0:
+            current_state = results[0].get("Status", "")
+            self.log.info(
+                "Current state of replication config(%s) is %s.", self.replication_config_arn, current_state
+            )
+            # replication must be deprovisioned before deleting
+            provision_status = self.hook.get_provision_status(
+                replication_config_arn=self.replication_config_arn
+            )
+
+            if (
+                current_state.lower() in self.VALID_STATES
+                and provision_status in self.TERMINAL_PROVISION_STATES
+            ):
+                self.log.info("DMS replication config(%s) is in valid state.", self.replication_config_arn)
+
+                self.hook.delete_replication_config(
+                    replication_config_arn=self.replication_config_arn,
+                    delay=self.waiter_delay,
+                    max_attempts=self.waiter_max_attempts,
+                )
+                self.handle_delete_wait()
```

Comment:
In order to simplify things I think you can remove that block of code. The `else` statement would also work if the replication config is in valid state

Reply:
Which block of code?

Reply:
The one I selected for this comment. From line `470`  to `483`